Userland/Libraries/LibGUI/Application.cpp

```diff
@@ -324,4 +325,73 @@ void Application::event(Core::Event& event)
     Object::event(event);
 }
 
+void Application::set_config_domain(String config_domain)
+{
+    m_config_domain = move(config_domain);
+}
+
+void Application::register_recent_file_actions(Badge<GUI::Menu>, Vector<NonnullRefPtr<GUI::Action>> actions)
+{
+    m_recent_file_actions = move(actions);
+    update_recent_file_actions();
+}
+
+void Application::update_recent_file_actions()
+{
+    size_t number_of_recently_open_files = 0;
+    auto update_action = [&](size_t index) {
+        auto& action = m_recent_file_actions[index];
```

Comment:
Two questions:
- Can the index be "out of bounds" here?
- Do we need to null check `action`?

Reply:
The index should never be out of bounds here, as we won't be running this update mechanism unless we first installed a recent files list in a menu.

If there's some kind of bug and we come here without first setting up the actions list, `Vector::operator[]` will panic with a `VERIFY` failure and it should be easy to track down what happened.